lib/GameObject/GameObject.h

```diff
@@ -3,24 +3,35 @@
 
 #include "Arduino.h"
 
-#ifndef _U8G2LIB_HH
+
 #include <U8g2lib.h>
-#endif
 
+
+/*
+    Classe di base su cui basare gli oggetti di gioco.
+    Estendendo questa classe si ha la possibiltà di gestire di usufruire di alcuni metodi relativi alla gestione delle cordinate.
+    Per la scelta di cosa disegnare a schermo si totalmente totalmente liberisi può gestire tramite metodi esterno o fancendo l'override del metodo disegna.
+*/
 class GameObject {
 private:
-    /* data */
+    /* attributi relativi alle cordinate e alle dimensioni dell'oggetto */
     int x;
     int y;
     int h;
     int b;
 public:
     GameObject(int,int,int,int);
-    virtual void disegna(U8G2 &t);
+    #ifndef _U8G2LIB_HH
```

Comment:
Non ho capito il #ifndef _U8G2LIB_HH, in questo caso la classe diventa un'interfaccia (metodo virtuale sotto) e richiama un metodo astratto che necessita di un classe che suppongo sia definita nella libreria U8G2LIB.
In ogni caso rimuoverei il parametro in ingresso e la lascerei completamente come interfaccia!

Reply:
e628ce6 - sistemato